BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/BetterShardsAPI.java

```diff
@@ -138,4 +138,23 @@ public static void randomSpawnPlayer(String server, UUID player) {
 	public static ConfigurationSection getConfigurationSection(JavaPlugin plugin, UUID uuid) {
 		return CustomWorldNBTStorage.getWorldNBTStorage().getConfigurationSection(uuid, plugin);
 	}
+	
+	/**
+	 * Registers a portal with BetterShards.
+	 * This should be called for every custom portal you have.
+	 * @param plugin_id The plugin specific id that your portal uses.
+	 * @param plugin The plugin that is registering this portal.
+	 * @param portal The portal Class.
+	 * @param name The name of the portalType that will show up to players.
+	 */
+	public static <E extends Portal> void registerPortal(int plugin_id, String plugin, Class<E> portal, String name) {
+		BetterShardsPlugin.getDatabaseManager().addPortalType(plugin_id, plugin);
```

Comment:
Given that this is called from startup and repeatedly elsewhere, shouldn't you be checking if this type exists _before_ adding it? Even though your uniqueness pkey guarantee will probably prevent a new row from being generated, relying on an error is risky... better to explicitly check imho.